### PR TITLE
Updating GeoFence documentation with example on CQL read filter limits on time range

### PR DESCRIPTION
**Pull request by dromagnoli**
Updating GeoFence documentation with examples on how to limit access to a specific layer by setting up a CQL Filter to only access a specific time range.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/extensions/geofence-server/gui.rst [hunk 1]
```diff
@@ -25,20 +25,36 @@ This page is displayed both when creating a new rule and modifying an existing r
...
+These make it possible to add additional filters to the layer for the rule in question. For example, the rule can alter the default style of the layer, specify which styles are available at all, which attributes are accessible for reading and/or writing, specify CQL filters for both reading and writing, specify a catalog mode, and an allowed area (WKT) filter.
 
 .. figure:: images/layerdetails.png
    :align: center
 
 
+A CQL Read Filter can be setup to allow users to only access a subset of the available data.
+
+For example, if you have a daily time series layer (a Layer with time attribute/dimension), 
+you can set a rule allowing users to only access data in a certain time range.
```
Comment (aaime) [inline doc/en/user/source/extensions/geofence-server/gui.rst:46]:
```suggestion
For example, having a daily time series layer (a Layer with time attribute/dimension), 
it's possible to restrict access to a given time range.
```

File: doc/en/user/source/extensions/geofence-server/gui.rst [hunk 2]
```diff
@@ -25,20 +25,36 @@ This page is displayed both when creating a new rule and modifying an existing r
...
+
+A CQL filter to limit access to the whole 2020 year (static time range) will be like this::
```
Comment (aaime) [inline doc/en/user/source/extensions/geofence-server/gui.rst:48]:
```suggestion
A CQL filter to limit access to the whole 2020 year (static time range) would look as follows::
```

File: doc/en/user/source/extensions/geofence-server/gui.rst [hunk 3]
```diff
@@ -25,20 +25,36 @@ This page is displayed both when creating a new rule and modifying an existing r
...
+
+   time between 2020-01-01 and 2020-12-31
+
+A CQL filter to only access data older than a week (dynamic time range) will be like this::
```
Comment (aaime) [inline doc/en/user/source/extensions/geofence-server/gui.rst:52]:
```suggestion
A CQL filter limiting access to data older than a week (dynamic time range) would look like this instead:
```

**Review by aaime — CHANGES_REQUESTED** [review]

**Comment by dromagnoli** [comment]
@aaime feedbacks applied.
thanks for the review

**Pull request merged**